### PR TITLE
Merge from upstream through 2020-05-28

### PR DESCRIPTION
**Pull request by amgleitman**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Merge facebook/react-native's main branch into react-native-macos's main branch through 2020-05-28 23:59:59 PDT (specifically, f438a6e8f60a5cd10b06f19bc5248f5805fbb739).

These changes were made by repeatedly running git merge <commit-id> and resolving merge conflicts as needed.

No major divergences were introduced, although one thing that I noticed was the introduction of a new method swizzling function in React/Base/RCTUtils.(h|m). The two previously existing functions return an `IMP` corresponding to the original method implementations in react-native-macos, while they return `void` upstream. I wasn't 100% sure what would be the best way to extend this pattern here, but since we're not using the result anyway at the moment, it's probably not a big deal. If it comes down to it, we can always create a separate issue for this.

## Validation

* yarn lint
* * ✖ 65 problems (0 errors, 65 warnings)
* yarn flow-check-ios
* yarn flow-check-macos
* yarn flow-check-android
* yarn test
* * Test Suites: 128 passed, 128 total
* * Tests: 1 skipped, 1247 passed, 1248 total
* * Snapshots: 573 passed, 573 total
* RNTester compiles on macOS and iOS, and no new regressions have been found

### CODE REVIEW

File: Libraries/Text/TextInput/RCTBaseTextInputViewManager.m [hunk 1]
```diff
@@ -58,6 +58,7 @@ @implementation RCTBaseTextInputViewManager
 RCT_EXPORT_VIEW_PROPERTY(blurOnSubmit, BOOL)
 RCT_EXPORT_NOT_OSX_VIEW_PROPERTY(clearTextOnFocus, BOOL) // TODO(macOS GH#774)
 RCT_REMAP_NOT_OSX_VIEW_PROPERTY(keyboardType, backedTextInputView.keyboardType, UIKeyboardType) // TODO(macOS GH#774)
+RCT_EXPORT_NOT_OSX_VIEW_PROPERTY(showSoftInputOnFocus, BOOL) // TODO(macOS GH#774)
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/RCTBaseTextInputViewManager.m:61]: I don't see this in our repo. Why does it have the macOS GH tag?

File: RNTester/RNTester/AppDelegate.mm [hunk 2]
```diff
@@ -161,6 +161,15 @@ - (void)loadSourceForBridge:(RCTBridge *)bridge
   _turboModuleManager = [[RCTTurboModuleManager alloc] initWithBridge:bridge
                                                              delegate:self
                                                             jsInvoker:bridge.jsCallInvoker];
+
+#if RCT_DEV
+  /**
+   * Eagerly initialize RCTDevMenu so CMD + d, CMD + i, and CMD + r work.
+   * This is a stop gap until we have a system to eagerly init Turbo Modules.
+   */
+  [_turboModuleManager moduleForName:"RCTDevMenu"];
```
Comment (HeyImChris) [inline RNTester/RNTester/AppDelegate.mm:170]: We probably want this in `RNTester_macOS/AppDelegate.mm` too. As a rule of thumb, any new changes to an iOS file should default to be made in the macOS version of the same file unless there's something iOS specific about it

File: React/Base/RCTUtilsUIOverride.h [hunk 3]
```diff
@@ -5,6 +5,13 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#import <Foundation/Foundation.h>
+#if TARGET_OS_OSX // TODO(macOS GH#774)
+#import <React/RCTUIKit.h>
+#else // [TODO(macOS GH#774)
+#import <UIKit/UIKit.h>
```
Comment (HeyImChris) [inline React/Base/RCTUtilsUIOverride.h:12]: We shouldn't need this. <React/RCTUIKit.h> should be pulling in UIKit.h already

File: Libraries/Text/TextInput/RCTBaseTextInputViewManager.m (hunk 1 above)
Comment (amgleitman) [reply Libraries/Text/TextInput/RCTBaseTextInputViewManager.m:61]: My understanding has been to add the GH tag whenever we introduce a divergence from upstream. Is my understanding correct, or is there something more to this?

File: RNTester/RNTester/AppDelegate.mm (hunk 2 above)
Comment (amgleitman) [reply RNTester/RNTester/AppDelegate.mm:170]: Added.

File: React/Base/RCTUtilsUIOverride.h (hunk 3 above)
Comment (amgleitman) [reply React/Base/RCTUtilsUIOverride.h]: Resolved.

File: Libraries/Text/TextInput/RCTBaseTextInputViewManager.m (hunk 1 above)
Comment (HeyImChris) [reply Libraries/Text/TextInput/RCTBaseTextInputViewManager.m:61]: That's spot on! Why do we need this divergence?

Comment (amgleitman) [reply Libraries/Text/TextInput/RCTBaseTextInputViewManager.m:61]: From what I've been able to tell, `showSoftInputOnFocus` seems to be a mobile-specific thing. It indicates whether or not the virtual keyboard appears when we focus on a text input, which doesn't seem to make sense on macOS as far as I'm aware.

**Review by HeyImChris — APPROVED** [review]

**Pull request merged**